### DISCUSSION TITLE
pvlib.shading.shaded_fraction1d

### DISCUSSION THREAD

**Discussion by kurt-rhee** [pvlib/pvlib-python#2303] [Q&A]
Hey @echedey-ls and @kandersolar, 

I just wanted to be absolutely sure, if you have a uniform array of trackers (unlimited sheds style) then is it unnecessary to project the solar zenith angle into rack coordinate system since the shading and the shaded trackers are parametrically identical?

**Comment by echedey-ls** [comment]
It was some time ago so I may be incorrect, but my interpretation of the projected solar zenith is that it calculates the slope of the plane between direct shade and direct sunlight, as the angle between its intersection with the horizontal plane and with the plane perpendicular to the rotation vector of a HSAT.

I know this answers no question, but I think the issue is with understanding what is being calculated. It is a must-know if you need 1D shade because it calculates the angle at which shade is being cast from the top of the shading tracker to the shaded one, in the shortest path possible. AFAIK, there is no other way to calculate shades parametrically without taking into account the PZSA.

*IIRC it's in the docstring, but this is equally valid for fixed-tracking, if it is considered a particular position of HSAT.

> is it unnecessary to project the solar zenith angle into rack coordinate system since the shading and the shaded trackers are parametrically identical?

`pvlib.shading.shaded_fraction1d` assumes the tracker casting the shadow has the same rotation angle as the shaded one by default, see second paragraph. I agree there is wording to improve in them, I need to prepare some PR for things like that...

Maybe this question helps you; did you consider any other path for shades calculation?


PS: @mikofski is a co-author on "Slope-Aware Backtracking for Single-Axis Trackers", he may also be able to help you.

**Reply by kandersolar** [reply]
> AFAIK, there is no other way to calculate shades parametrically without taking into account the PZSA.

I agree.  Solar zenith alone certainly isn't sufficient; consider a typical SAT array with `axis_azimuth=180` and `solar_elevation=5`.  When `solar_azimuth=90`, there will be significant potential for shading.  When `solar_azimuth=180`, shading isn't possible.
